haptools/__main__.py

```diff
@@ -498,3 +498,60 @@ def transform(
 if __name__ == "__main__":
     # run the CLI if someone tries 'python -m haptools' on the command line
     main(prog_name="haptools")
+
+
+
+
+############ Haptools index ###############
+
+@main.command(short_help="Sort and index .hap files")
+@click.argument("haplotypes", type=click.Path(exists=True, path_type=Path))
+@click.option(
+    "-o",
+    "--output",
+    type=click.Path(path_type=Path),
+    default= None,
+    show_default="stdout",
+    help="A .hap file containing sorted and indexed haplotypes and variants",
+)
+@click.option(
+    "-v",
+    "--verbosity",
+    type=click.Choice(["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG", "NOTSET"]),
+    default="ERROR",
+    show_default="only errors",
+    help="The level of verbosity desired",
+)
+
+def index(
+    haplotypes: Path,
+    output: Path = None,
+    verbosity: str = 'CRITICAL',
+):
+
+    """
+    
+        Takes in an unsorted .hap file and outputs it as a .gz and a .tbi file
+
+        Parameters
```

Comment:
I've recently started removing sections within the docstrings in functions in `__main__.py` like "Parameters", since click already enables automated documentation through the parameters you defined, anyway

so you can probably remove most of this docstring but keep the part that says
```
Takes in an unsorted .hap file and outputs it as a .gz and a .tbi file
```